flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/BucketFactory.java

```diff
@@ -37,6 +39,7 @@
 			final long initialPartCounter,
 			final BucketWriter<IN, BucketID> bucketWriter,
 			final RollingPolicy<IN, BucketID> rollingPolicy,
+			@Nullable final FileLifeCycleListener<IN, BucketID> fileListener,
```

Comment:
Why not having an `Optional` instead of a `@Nullable`? I think `null` leads to more "fragile" code and more difficult to maintain.

Reply:
https://flink.apache.org/contributing/code-style-and-quality-java.html#java-optional
In Flink code style, it is recommended to use the `Optional` only in method return values.